src/PowerShellEditorServices.Protocol/MessageProtocol/MessageWriter.cs

```diff
@@ -104,10 +104,20 @@ public async Task WriteMessageAsync(Message messageToWrite)
             // message loop doesn't get blocked while waiting for I/O to complete.
             using (await this.writeLock.LockAsync())
             {
-                // Send the message
-                await this.outputStream.WriteAsync(headerBytes, 0, headerBytes.Length);
-                await this.outputStream.WriteAsync(messageBytes, 0, messageBytes.Length);
-                await this.outputStream.FlushAsync();
+                try
+                {
+                    // Send the message
+                    await this.outputStream.WriteAsync(headerBytes, 0, headerBytes.Length);
+                    await this.outputStream.WriteAsync(messageBytes, 0, messageBytes.Length);
+                    await this.outputStream.FlushAsync();
+                }
+                catch (Exception e) when (
+                    e is ObjectDisposedException ||
+                    e is IOException)
+                {
+                    // We catch this exception for when the DebugAdapter disconnects while still processing a message.
+                    logger.WriteException("Tried to write to the output stream but it was already closed & broken.", e);
```

Comment:
This probably happens decently often, do we want to make this a verbose message? If you can't include an exception in the verbose message then never mind.

Reply:
verbose doesn't allow an exception but I did change this to `WriteHandledException`!